### ISSUE TITLE
Wrong parent work with parentwork plugin

### ISSUE THREAD

**Issue by aereaux** [beetbox/beets#4001]
I have [this album](https://beta.musicbrainz.org/release/dd440e84-add3-4f5b-b762-602d8df1c643) in my collection.  On CD the tracks 4.3 to 4.8 get their parentwork as [Fantasia](https://beta.musicbrainz.org/work/74e65a9e-608d-4097-b238-c1bb6536ce7c), instead of the proper [ballet suite](https://beta.musicbrainz.org/work/fd337fdc-a511-4830-8ee7-09d4ada32a54).  I'm not sure if there's a good way to fix this either in the code or in the data in musicbrainz.  One thought I had was to choose the work with the earliest composition date, but that might not always work (especially if the dates are not specified), and might be difficult to implement, including needing to get more data than we already do.

If anyone has any other ideas about how to work around or solve this problem, that would be great.

**Comment by sampsyo** [comment]
That's odd! @dosoe, would you mind taking a look?

**Comment by aereaux** [comment]
Thanks for the quick response.  It's not so much that it's odd as that it's a problem with the way the Fantasia soundtrack is categorized in MB.  I think right now there's not much logic when a work has multiple parents, so it appears to just take the first one:
![image](https://user-images.githubusercontent.com/8238401/125120743-6cf29500-e0e2-11eb-9f48-d3ec28c2f403.png)  I'm mostly just not sure how that could be improved.

**Comment by dosoe** [comment]
Yes, that's exactly what's happening. One thing I would like to do is allow to choose a parentwork when there are several, that's been a problem for me as well. I'm gonna do it at some point, but I'm finishing a PR for performers right now.

**Comment by aereaux** [comment]
Do you think adding some heuristics would improve this?  At least in this case preferring other types of works over soundtracks would help and seems helpful in the general case.

**Comment by dosoe** [comment]
Actually, on one of my forks I hardcoded a work to be taken as a part of a bach cantata rather than of the neon genesis evangelion soundtrack, but I don't like this solution, because it assumes noone will listen to it as part of the soundtrack. Also, I have a few soundtracks in my collection, and I'm not sure all soundtracks are marked as such on MusicBrainz. I think making a choice when getting the parentwork if there are several is the good solution, but I have to do it, I think I will do it before the end of the year for sure, maybe earlier. I mean, it shouldn't be too hard, just take a plugin that offers you a choice as a template or check the documentation, it's explained there.
